src/Relude/Extra/CallStack.hs

```diff
@@ -0,0 +1,26 @@
+{- | Contains useful functions to work with GHC callstack.
+-}
+
+module Relude.Extra.CallStack
+       ( ownName
+       ) where
+
+import Relude
+
+import GHC.Stack (HasCallStack, callStack, getCallStack)
+
+-- TODO: better name?
+{- | This function returns name of the caller for this function. But it requires
+for caller to have 'HasCallStack' constraint. Otherwise it returns @"ownName".
```

Comment:
Better phrasing here would be:
`This function returns the name of its caller function, but its requires that the caller function has 'HasCallStack' constraint. Otherwise, it returns @"ownName"@.`